docs/proposals/202105256-cluster-class-and-managed-topologies.md

```diff
@@ -261,8 +468,20 @@ type LocalObjectTemplate struct {
 	  // for the cluster.
 	  // +optional
 	  Workers *WorkersTopology `json:"workers,omitempty"`
+    
+      // Variables can be used to customize the ClusterClass through
+	  // patches. They must comply to the corresponding
+	  // VariableDefinitionClasses defined in the ClusterClass.
+	  // +optional
+	  Variables []VariableTopology `json:"variables,omitempty"`
     }
```

Comment:
Have you considered to also allow Patches on individual clusters?

While the goal with ClusterClass obviously is to create clusters that are as homogenous as possible, it might be useful to be able to apply patches to individual clusters that weren't considered when creating the Class. Especially applying hotfixes to individual clusters would be a good use case for this, as it isn't possible to modify any of the cluster resources when they are managed by CAPI.

Reply:
I think this use case is pretty close to what could be achieved with optional patches (which is not part of the current iteration). The idea of optional patches is that you can opt-in into patches of the ClusterClass in the Cluster resources.

So the workflow for hotfixes would be:
* Add a new optional patch to the ClusterClass
* Enable the optional patch on a per-cluster basis

Imho we should try to avoid introducing the whole complexity of patches on the Cluster resource.

Reply:
I adjusted the note about optional patches a bit
```
**Note**: We are considering to add a field to allow defining optional patches. This would allow
adding optional patches in the ClusterClass and then activating them on a per-cluster basis via opt-in.
```